### PR TITLE
feat: implement header component

### PR DESCRIPTION
**Pull request by BogdanDanilescu**
### **Header implementation for jinja/nunjucks library**

Key aspects:

- Properties added ( nav links, secondary nav links )
- 3 Different views depending on the screen size ( desktop, tablet, mobile )
- Implementation example in wagtail application
- Compatible in case of no JS

Screenshots
_**Desktop view:**_
image
**_Tablet view_**
image
**_Mobile view_**
image

### CODE REVIEW

File: packages/html/ds/src/header/components/headerDesktop.html [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{% from 'header/assets/logoIcon.html' import LogoIcon %} 
+{% from 'header/assets/searchIcon.html' import searchIcon %} 
+{% from 'header/components/headerSearchContainer.html' import HeaderSearchContainer %}
+
+{% macro govieHeaderDesktop(props) %}
+
+<header class="sm:gi-flex gi-hidden gi-flex-col gi-relative gi-w-full">
+  {% if props.secondaryNavLinks %}
```
Comment (pregno) [inline packages/html/ds/src/header/components/headerDesktop.html:8]:
```suggestion
  {% if props.languages %}
```

File: packages/html/ds/src/header/components/headerDesktop.html [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  <div class="gi-bg-emerald-700">
+    <ul
+      class="gi-flex gi-justify-end gi-gap-xl gi-items-center gi-h-full gi-mr-3xl gi-my-1"
+    >
+      {% for link in props.secondaryNavLinks %}
```
Comment (pregno) [inline packages/html/ds/src/header/components/headerDesktop.html:13]:
```suggestion
      {% for link in props.languages %}
```

File: packages/html/ds/src/header/components/headerMenu.html [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    {% for link in props.navLinks %}
+    <li>
+      <a
+        href="{{ link.href }}"
+        class="hover:gi-underline hover:gi-underline-offset-sm gi-block gi-py-4 gi-border-gray-100 gi-border-b gi-border-solid"
+      >
+        <span class="gi-font-bold gi-text-sm">{{ link.label }}</span>
+      </a>
+    </li>
+    {% endfor %} {% for link in props.secondaryNavLinks %}
```
Comment (pregno) [inline packages/html/ds/src/header/components/headerMenu.html:31]:
```suggestion
    {% endfor %} {% for link in props.languages %}
```

**Review by pregno — CHANGES_REQUESTED** [review]

**Review by pregno — APPROVED** [review]

File: examples/wagtail/core/jinja/vars.jinja [hunk 4]
```diff
@@ -38,4 +38,32 @@
     },
   ],
 }
+%}
+
+{% set HeaderProps = {
```
Comment (hamza14khan) [inline examples/wagtail/core/jinja/vars.jinja:43]: Can you please fix these changes with tabs 2 spacing? Thanks!

File: packages/html/ds/src/header/header.html [hunk 5]
```diff
@@ -1,33 +1,11 @@
...
-          Search
-        </button>
-      </div>
-    </div>
-  </div>
-</header>
-{% endmacro %}
+
+<div class="gi-relative" data-module="gieds-header">
+    {{ govieHeaderDesktop(props) }}
```
Comment (hamza14khan) [inline packages/html/ds/src/header/header.html:7]: Same as above.

Comment (BogdanDanilescu) [reply packages/html/ds/src/header/header.html:7]: fixed

Comment (BogdanDanilescu) [reply packages/html/ds/src/header/header.html:7]: In this case I left a space intentionally for readability, I think it looks better like this
image

This way we can see clearly and beginning and end of the macro

**Pull request merged**